source/adios2/toolkit/transportman/TransportMan.cpp

```diff
@@ -421,5 +434,14 @@ void TransportMan::CheckFile(
     }
 }
 
+void TransportMan::WaitForAsync() const
+{
+    // Ensure any prior OpenFilesAsync is complete
+    if (m_FutureOpenFiles.valid())
+    {
+        m_FutureOpenFiles.get();
+    }
```

Comment:
Can the functions just call `m_FutureOpenFiles.wait()` instead?

Reply:
`wait()` also requires `valid()` to be true.

Reply:
Then at least `wait` instead of `get`?

Reply:
I think consuming the result with get() is the thing to do.  If get() sets valid to false, then it's better for every call after the first to fail on valid() and not have to call wait() too.   Impact for multiple calls to OpenFilesAsync too?